tests/unit_tests/Objects/variable/test_parameter.py

```diff
@@ -91,10 +113,326 @@ def test_init_value_max_exception(self):
+        assert dependent_parameter.dependency_expression == '2*a'
+        assert dependent_parameter.dependency_map == {'a': normal_parameter}
+        assert dependent_parameter.name == 'dependent'
+        assert dependent_parameter.display_name == 'display_name'
+        self.compare_parameters(dependent_parameter, 2*normal_parameter)
+
+        # Then
+        normal_parameter.value = 2
+
+        # Expect
+        self.compare_parameters(dependent_parameter, 2*normal_parameter)
+
+    def test_dependent_parameter_with_unique_name(self, clear, normal_parameter: Parameter):
+        # When Then
+        dependent_parameter = Parameter.from_dependency(
+            name = 'dependent', 
+            dependency_expression='2*"Parameter_0"',
+        )
+
+        # Expect
+        assert dependent_parameter.dependency_expression == '2*"Parameter_0"'
+        assert dependent_parameter.dependency_map == {'__Parameter_0__': normal_parameter}
+        self.compare_parameters(dependent_parameter, 2*normal_parameter)
+
+        # Then
+        normal_parameter.value = 2
+
+        # Expect
+        self.compare_parameters(dependent_parameter, 2*normal_parameter)
+
+    def test_process_dependency_unique_names_double_quotes(self, clear, normal_parameter: Parameter):
+        # When
+        independent_parameter = Parameter(name="independent", value=1, unit="m", variance=0.01, min=0, max=10, unique_name='Special_name')
+        normal_parameter._dependency_map = {}
+
+        # Then
+        normal_parameter._process_dependency_unique_names(dependency_expression='2*"Special_name"')
+
+        # Expect
+        assert normal_parameter._dependency_map == {'__Special_name__': independent_parameter}
+        assert normal_parameter._clean_dependency_string == '2*__Special_name__'
+
+    def test_process_dependency_unique_names_single_quotes(self, clear, normal_parameter: Parameter):
+        # When
+        independent_parameter = Parameter(name="independent", value=1, unit="m", variance=0.01, min=0, max=10, unique_name='Special_name')
+        independent_parameter_2 = Parameter(name="independent_2", value=1, unit="m", variance=0.01, min=0, max=10, unique_name='Special_name_2')
+        normal_parameter._dependency_map = {}
+
+        # Then
+        normal_parameter._process_dependency_unique_names(dependency_expression="'Special_name' + 'Special_name_2'")
+
+        # Expect
+        assert normal_parameter._dependency_map == {'__Special_name__': independent_parameter, 
+                                                    '__Special_name_2__': independent_parameter_2}
+        assert normal_parameter._clean_dependency_string == '__Special_name__ + __Special_name_2__'
+
+    def test_process_dependency_unique_names_exception_unique_name_does_not_exist(self, clear, normal_parameter: Parameter):
+        # When
+        normal_parameter._dependency_map = {}
+
+        # Then Expect
+        with pytest.raises(ValueError):
```

Comment:
```suggestion
        with pytest.raises(ValueError) as e: 
```
and then test the error message with `assert 'TEST string' in str(e)` perhaps

Reply:
Ah I see one can do it cleaner with `pytest.rases(..., match=)` as in your next test.

Reply:
:D Do you think I should include the string match in this test?

Reply:
Yes, I think it is nice to ensure that the error one expects is actually being thrown.